WooCommerce/Classes/ViewRelated/Orders/Order Details/Address Edit/AddressMapPickerViewModel.swift

```diff
@@ -0,0 +1,190 @@
+            } else {
+                fields.state = address.state
+            }
+        } else {
+            fields.country = address.isoCountryCode
+            fields.state = address.state
+        }
+    }
+}
+
+@available(iOS 17, *)
+private extension AddressMapPickerViewModel {
+    func configureSearchCompleter() {
+        searchCompleter.resultTypes = .address
+        searchCompleter.delegate = self
+    }
+
+    func configureMap(with fields: AddressFormFields) {
+        // If fields are empty and we have location permission, use current location for the initial map region.
+        if fields.address1.isEmpty && fields.address2.isEmpty && fields.city.isEmpty {
+            let locationManager = CLLocationManager()
+            if locationManager.authorizationStatus == .authorizedWhenInUse ||
+                locationManager.authorizationStatus == .authorizedAlways,
+               let currentLocation = locationManager.location {
+                DispatchQueue.main.async { [weak self] in
+                    guard let self else { return }
+                    region = MKCoordinateRegion(
+                        center: currentLocation.coordinate,
+                        span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
+                    )
+                }
+                return
+            }
+        }
+
+        // Tries to geocode the existing address if possible.
+        let addressString = formatAddressString(fields: fields)
+        if addressString.isNotEmpty {
+            Task { @MainActor in
+                do {
+                    let placemarks = try await CLGeocoder().geocodeAddressString(addressString)
+                    guard let location = placemarks.first?.location else {
+                        return
+                    }
+
+                    region = MKCoordinateRegion(
+                        center: location.coordinate,
+                        span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
+                    )
+                    annotations = [MapAnnotation(coordinate: location.coordinate)]
+                } catch {
+                    DDLogError("⛔️ Error geocoding initial address: \(error)")
+                }
+            }
+        }
+    }
+
+    @MainActor
+    func onSelectedPlacemark(_ placemark: MKPlacemark, result: MKLocalSearchCompletion) async {
+        await withCheckedContinuation { continuation in
+            withAnimation { [weak self] in
+                guard let self else { return }
+                searchResults = []
+                region = MKCoordinateRegion(
+                    center: placemark.coordinate,
+                    span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
+                )
+                annotations = [MapAnnotation(coordinate: placemark.coordinate)]
+                selectedPlace = placemark
+                continuation.resume()
+            }
```

Comment:
Is it supposed to be using a version with completion block? If the intention is to wait for the animation to be completed?

```suggestion
    @MainActor
        await withCheckedContinuation { continuation in
            withAnimation {
                searchResults = []
                region = MKCoordinateRegion(
                    center: placemark.coordinate,
                    span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
                )
                annotations = [MapAnnotation(coordinate: placemark.coordinate)]
                selectedPlace = placemark
            } completion: {
                continuation.resume()
            }
        }
```

Reply:
TIL about the completion block in `withAnimation`, updated in https://github.com/woocommerce/woocommerce-ios/pull/15958/commits/465144a75461c455fcd5218999985edd7dbb5c5c.